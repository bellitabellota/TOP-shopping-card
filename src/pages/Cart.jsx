import NavBar from "../sections/NavBar";

function Cart() {
  return(
    <>
      <main>
        Cart with items
      </main>
    </>
  )
}

export default Cart;