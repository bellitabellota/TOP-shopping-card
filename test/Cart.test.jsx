import Cart from "../src/pages/Cart";
import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { useOutletContext } from "react-router-dom";
import { beforeEach } from "vitest";

vi.mock("react-router-dom", () => ({
  useOutletContext: vi.fn(),
}));

describe("Cart", () => {
  beforeEach(() => {
    const selectedProductsMock = [{
      id: 5,
      title: "Selected Products 1",
      price: 20.00,
      description: "The Selected Products 1 description ...",
      category: "selected products",
      image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
      rating: {
        rate: 3.9,
        count: 120
      },
      quantity: 2
    },
    {
      id: 9,
      title: "Selected Products 2",
      price: 10.00,
      description: "The Selected Products 2 description ...",
      category: "selected products",
      image: "https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg",
      rating: {
        rate: 4.1,
        count: 259
      },
      quantity: 5
    }]

    useOutletContext.mockReturnValue([null, null, selectedProductsMock]);
  });

  it("renders a page with the selected product", () => {

    const {container} = render(<Cart />);

    expect(container).toMatchSnapshot();
  })

  it("displays the total of the selected products", () => {
    render(<Cart />)
    expect(screen.getByText("Total: 90")).toBeInTheDocument();

  })
})