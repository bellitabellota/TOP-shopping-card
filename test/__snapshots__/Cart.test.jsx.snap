// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart > renders a page with the selected product 1`] = `
<div>
  <h1>
    Your cart
  </h1>
  <div
    class="_selectedProductContainer_27e39d"
  >
    <div
      class="_selectedProductContainerDescription_27e39d"
    >
      <p
        class="_selectedProductTitleDescription_27e39d"
      >
        Product
      </p>
      <p
        class="_selectedProductQuantityDescription_27e39d"
      >
        Quantity
      </p>
      <p
        class="_selectedProductSubtotalDescription_27e39d"
      >
        Subtotal
      </p>
    </div>
    <div
      class="_selectedProduct_27e39d"
    >
      <p
        class="_selectedProductTitle_27e39d"
      >
        Selected Products 1
      </p>
      <p
        class="_selectedProductQuantity_27e39d"
      >
        2
      </p>
      <p
        class="_selectedProductSubtotal_27e39d"
      >
        40
         €
      </p>
    </div>
    <div
      class="_selectedProduct_27e39d"
    >
      <p
        class="_selectedProductTitle_27e39d"
      >
        Selected Products 2
      </p>
      <p
        class="_selectedProductQuantity_27e39d"
      >
        5
      </p>
      <p
        class="_selectedProductSubtotal_27e39d"
      >
        50
         €
      </p>
    </div>
    <div
      class="_totalAmount_27e39d"
    >
      Total: 
      90
    </div>
  </div>
</div>
`;
